docs: translate blogs on 09/2021 (#678)
iresty/iresty.github.io: website/blog/2021/09/09/how-to-contribute-to-an-OpenSource-without-coding.md

---
title: How can I contribute to an open source project without writing code?
author: Ruofei Zhao
authorURL: "https://github.com/Serendipity96"
authorImageURL: "https://avatars.githubusercontent.com/u/23514812?v=4"
keywords: 
- Open Source
- Contribution
description: This article introduces many ways to contribute to open source projects without writing code. You can participate in open source projects by writing articles, making videos, sharing them externally, building local communities, and actively participating in community discussions.
tags: [Community]
---

> This article describes many ways to contribute to open source projects without writing code. You can get involved in open source projects by writing articles, making videos, sharing them externally, building local communities, and participating actively in community discussions.

> <! --truncate -->

Reviewer: ```suggestion
<!--truncate-->
```

> I have contributed several PRs and submitted several issues to open source projects, and I still remember clearly the excitement when my first PR was merged a few years ago, and I experienced the fun of participating in open source. But only in the contributor stage, no more, because I later do not do R & D, **think not familiar with the code, can not fix the bug will not have the opportunity to contribute.**

Reviewer: ```suggestion
I have contributed several PRs and submitted several issues to open source projects, and I still remember clearly the excitement when my first PR was merged a few years ago, and I experienced the fun of participating in open source. But my experience remains in the contributor stage with no further progress, because I changed my position from front-end developer to marketing. I am afraid that I would be less familiar with the code,  and I can not fix the bug, which will result in no contribution to the project.
```

There is a misunderstanding here, not only contributing code and fixing bugs are contributions to the project. Is the only way to become a contributor is to contribute code? **If I can't write code and I'm not a developer, how can I become a contributor to an open source project and even get promoted to committer?** If I don't know how to write code and am not a developer, how can I become a contributor to an open source project or even be promoted to committer?

After I sorted it out, I found that there are many ways to become a contributor, so I'll share them here. (The author is also on the way to practice)

> ## Non-code way

Reviewer: ```suggestion
## Contributing without Coding
```

#### is active on Github

Github is the most common site for developers, and most open source projects are hosted on Github. I can

- raise an issue with my thoughts on how to improve the project
- submit a bug report
- review pr submitted by other partners

These are all ways to contribute to open source projects.

#### Apache mail list vs Github

Why use a mailing list when you have Github? The code is hosted on Github, and it seems like all the community collaboration is done on Github?

If there are no objections to discussing an issue on the mailing list, the proposal will be approved by default. If there are objections, they will be fully discussed and eventually agreed or set aside.

The use of mailing lists for discussion is also a way to identify with the Apache culture. For projects that follow Apache rules, the mailing list is the primary place for the community, and Github is a convenient collaborative tool.

> ## Code approach

Reviewer: ```suggestion
## Contributing with Coding
```
